freedomofpress/securedrop-client
# Top Pane and User Button

## Description

Resolves https://github.com/freedomofpress/securedrop-client/issues/279
Resolves https://github.com/freedomofpress/securedrop-client/issues/298

## Summary of changes

#### UX Review needed

* [Behavior] Now, instead of having to click directly on the dropdown icon, a user can click anywhere on the `UserButton` to open the menu.

* [Behavior] Instead of hiding the refresh icon when "offline" disable it to show disabled image. 

* [Behavior] Added Static & Active states to the refresh icon, see freedomofpress/securedrop-ux#17 (comment)

* [Behavior] Before error messages were not displayed to the user. Now they show up in the center of the top pane for 10 seconds or until replaced with a new error message.

* [Styling] Added text styles for Refresh timestamp, journalist username, and sign-in button, see https://app.zeplin.io/project/5c807ea562f734bd2756b243/dashboard

* [Styling] Added a vertical bar and error icon next to the error status message.

#### Other changes

* Added background gradients on the top pane, left pane, and sign-in button as placeholders until final design is finished.

## Test Plan

  * Click on journalist name to see drop-down menu pop up
  * Click on drop-down arrow to see drop-down menu pop up
  * When signed out, click on star or delete icons to see 10-second error message centered in top pane
  * Test that error pop hides when a message is cleared because of logging in. To do this, hit the star or delete icons to trigger the error message, then log in before 10 seconds is up to see error pop up go away.
  * When signed in, click on refresh and see active-state refresh icon
  * Sign out and see refresh icon change to the disabled image

## A little demo

![demo3](https://user-images.githubusercontent.com/4522213/55387758-df797100-54e6-11e9-91a1-1869c0e4491e.gif)
